lib/composables/Table.ts

```diff
@@ -4,6 +4,10 @@ import { reactive } from 'vue'
 import type { Day } from '../support/Day'
 import type { DropdownSection } from './Dropdown'
 
+type Optional<T> = {
```

Comment:
Let's make this public and name `TableSummary` to be consistent with other stuff like `TableColumns`.

Just so that it looks good when listing this type in the docs 👍 (Although we don't have a docs for this composable yet but for the future 👀

Reply:
What should be the `TableSummary`?

```ts
export type TableSummary<R> = Optional<R>
```

This? Or did you mean change Optional to TableSummary:

```ts
export type TableSummary<T> = {
  [P in keyof T]?: T[P] | null | undefined
}
```

Reply:
Either way is fine! But for now, The later is what I was thinking 👍 So this.

```ts
export type TableSummary<T> = {
  [P in keyof T]?: T[P] | null | undefined
}
```

Reply:
Done! 🙌

Reply:
🎉